libs/sdk-ui-dashboard/src/model/layout/sizing.ts

```diff
@@ -0,0 +1,136 @@
+// (C) 2019-2022 GoodData Corporation
+
+import {
+    AnalyticalWidgetType,
+    IInsight,
+    IInsightDefinition,
+    IKpi,
+    ISettings,
+    isInsight,
+    isKpi,
+    isKpiWidget,
+    isKpiWithoutComparison,
+    IWidget,
+    widgetType,
+} from "@gooddata/sdk-model";
+import {
+    fluidLayoutDescriptor,
+    getInsightSizeInfo,
+    INSIGHT_WIDGET_SIZE_INFO_DEFAULT,
+    INSIGHT_WIDGET_SIZE_INFO_DEFAULT_LEGACY,
+    IVisualizationSizeInfo,
+    KPI_WIDGET_SIZE_INFO_DEFAULT,
+    KPI_WIDGET_SIZE_INFO_DEFAULT_LEGACY,
+} from "@gooddata/sdk-ui-ext";
+
+import { ObjRefMap } from "../../_staging/metadata/objRefMap";
+
+/**
+ * @internal
+ */
+export type MeasurableWidgetContent = IInsightDefinition | IKpi;
+
+export const getSizeInfo = (
+    settings: ISettings,
+    widgetType: AnalyticalWidgetType,
+    widgetContent?: MeasurableWidgetContent,
+): IVisualizationSizeInfo => {
+    if (widgetType === "kpi") {
+        return getKpiSizeInfo(settings, widgetContent);
+    }
+
+    return getVisualizationSizeInfo(settings, widgetContent);
+};
+
+const getVisualizationSizeInfo = (
+    settings: ISettings,
+    insight?: MeasurableWidgetContent,
+): IVisualizationSizeInfo => {
+    let sizeInfo;
+    if (isInsight(insight)) {
+        sizeInfo = getInsightSizeInfo(insight, settings);
+    }
+
+    if (!sizeInfo) {
+        if (!settings.enableKDWidgetCustomHeight) {
+            return INSIGHT_WIDGET_SIZE_INFO_DEFAULT_LEGACY;
+        }
+        return INSIGHT_WIDGET_SIZE_INFO_DEFAULT;
+    }
+    return sizeInfo;
+};
+
+const getKpiSizeInfo = (settings: ISettings, kpi?: MeasurableWidgetContent): IVisualizationSizeInfo => {
+    if (!settings.enableKDWidgetCustomHeight) {
+        return KPI_WIDGET_SIZE_INFO_DEFAULT_LEGACY;
+    }
+    if (!isKpi(kpi)) {
```

Comment:
Why is this condition here? isKpi can be false in getKpiSizeInfo?

Reply:
Or is it for placeholders?

Reply:
indeed, I have no idea, this function is moved from layout sizing utils without changes https://github.com/gooddata/gooddata-ui-sdk/pull/2655/files#diff-2fab94c22a67fb77aa71b00ad301467efa1f26c437a974e82c2a364b8e7854f2L484